File: Java_projet_perso_/src/main/java/Descartes.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.util.concurrent.TimeUnit;

public class Descartes extends Traitement{
	public int iteration=0;
	public float pointA;
	public float pointB;
	public float resultat;
	
	public Descartes() {
		
	}

	public void setFonction(Fonction fx) {
		this.f = new Fonction();
		this.f = fx;
	}

	public void setPointA(float i) {
		this.pointA = i;
	}

	public void setPointB(float i) {
		this.pointB = i;
	}

	public void resolution(String path) throws FileNotFoundException {
		try {
			this.methode="Descartes";
			PrintWriter fd = new PrintWriter(new File(path));
			float r=0;
			float a=this.pointA,b=this.pointB;
			float tol=(float) 0.005;
			do{
				r =(float) (b-this.f.f(b)* ((b-a) / (this.f.f(b)-this.f.f(a)) ));
				a=b;
				b=r;
				fd.println(r+" "+this.f.f(r));
				this.iteration++;
			}while((Math.abs(this.f.f(r))>tol) && (b!=a) && (this.iteration<10000));
			if(this.iteration>=10000)this.iteration=-1;
			if(b==a)this.iteration=-2;
			this.resultat=r;
			this.pathDatas=path;	
			fd.close();
		}
		catch(Exception e) {
			e.getStackTrace();
			this.pathDatas="null";
		}
	}
}

File: codeigniter/app/Libraries/Descartes.php
<?php
namespace App\Libraries;

class Descartes extends Traitement {
    public $iteration = 0;
    public $pointA;
    public $pointB;
    public $resultat;
  
    public function __construct() {
      
    }  

    public function setPointA(float $i) {
      $this->pointA = $i;
    }
  
    public function setPointB(float $i) {
      $this->pointB = $i;
    }
  
    public function resolution(string $path) {
      $this->methode = "Descartes";
      try {
          $outputFile = fopen($path, "w");
          if (!$outputFile) {
            throw new Exception("Failed to open file: " . $path);
          }
        $r = 0;
        $a = $this->pointA;
        $b = $this->pointB;
        $tol = 0.005;
  
        while ((float)abs($this->f->f($r)) > $tol && $b !== $a && $this->iteration < 10000) {
          $r = (float)($b - ($this->f->f($b) * (($b - $a) / ($this->f->f($b) - $this->f->f($a)))));
          $a = $b;
          $b = $r;
          fwrite($outputFile, $r . " " . $this->f->f($r) . PHP_EOL);
          $this->iteration++;
        }
  
        if ($this->iteration >= 10000) {
          $this->iteration = -1;
        } 
        else if ($b === $a) {
          $this->iteration = -2;
        }
  
        $this->resultat = $r;
        $this->pathDatas = $path;
        fclose($outputFile);
      } catch (Exception $e) {
        echo $e->getMessage() . PHP_EOL;
        $this->pathDatas = "null";
      }
    }
  }
?>